<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Login</class>
 <widget class="QDialog" name="Login">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>400</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>聊天室登录系统</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* 渐变背景 */
                QDialog {
                background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
                stop:0 #2c3e50, stop:1 #1a2a6c);
                border-radius: 12px;
                border: 1px solid #3498db;
                }

                /* 暗色模式优化 */
                @media (prefers-color-scheme: dark) {
                QDialog {
                background: qlineargradient(x1:0, y1:0, x2:1, y2:1,
                stop:0 #1a1a2e, stop:1 #16213e);
                }
                }</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="2" column="1" colspan="2">
    <widget class="QComboBox" name="userComboBox">
     <property name="styleSheet">
      <string notr="true">QComboBox {
                            height: 36px;
                            border: 1px solid rgba(255,255,255,0.3);
                            border-radius: 6px;
                            padding: 0 12px;
                            background: rgba(255, 255, 255, 0.15);
                            color: white;
                            font-size: 14px;
                            }
                            
                            QComboBox QAbstractItemView {
                            background: #2c3e50;
                            color: white;
                            border: 1px solid #3498db;
                            }
                            QComboBox QAbstractItemView::item {
                            padding: 6px;
                            }</string>
     </property>
     <property name="editable">
      <bool>true</bool>
     </property>
     <property name="placeholderText">
      <string>请输入账号</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="3">
    <widget class="QFrame" name="line">
     <property name="styleSheet">
      <string notr="true">background: rgba(255,255,255,0.2);</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::HLine</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Sunken</enum>
     </property>
     <property name="lineWidth">
      <number>2</number>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="userLabel">
     <property name="styleSheet">
      <string notr="true">font-size: 15px;
                            font-weight: bold;
                            color: rgba(255, 255, 255, 0.9);</string>
     </property>
     <property name="text">
      <string>账 号：</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1" colspan="2">
    <layout class="QHBoxLayout" name="buttonLayout">
     <item>
      <widget class="QPushButton" name="loginButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>40</height>
        </size>
       </property>
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton {
                                    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
                                    stop:0 #3498db, stop:1 #2980b9);
                                    color: white;
                                    border-radius: 6px;
                                    padding: 8px 20px;
                                    font-size: 15px;
                                    font-weight: bold;
                                    border: none;
                                    }
                                    QPushButton:hover {
                                    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
                                    stop:0 #2980b9, stop:1 #2573a7);
                                    }
                                    QPushButton:pressed {
                                    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
                                    stop:0 #2573a7, stop:1 #1d6fa5);
                                    }</string>
       </property>
       <property name="text">
        <string>登 录</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="registerButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>40</height>
        </size>
       </property>
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton {
                                    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
                                    stop:0 #2ecc71, stop:1 #27ae60);
                                    color: white;
                                    border-radius: 6px;
                                    padding: 8px 20px;
                                    font-size: 15px;
                                    font-weight: bold;
                                    border: none;
                                    }
                                    QPushButton:hover {
                                    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
                                    stop:0 #27ae60, stop:1 #219653);
                                    }
                                    QPushButton:pressed {
                                    background: qlineargradient(x1:0, y1:0, x2:1, y2:0,
                                    stop:0 #219653, stop:1 #1e8449);
                                    }</string>
       </property>
       <property name="text">
        <string>注 册</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="1" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QCheckBox" name="rememberCheck">
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="styleSheet">
        <string notr="true">font-size: 14px;
                                    color: rgba(255, 255, 255, 0.9);
                                    QCheckBox::indicator {
                                    width: 16px;
                                    height: 16px;
                                    border: 1px solid #3498db;
                                    border-radius: 3px;
                                    background: rgba(255, 255, 255, 0.1);
                                    }
                                    QCheckBox::indicator:checked {
                                    background: #3498db;
                                    border: 1px solid #2980b9;
                                    }</string>
       </property>
       <property name="text">
        <string>记住密码</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="autoLoginCheck">
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="styleSheet">
        <string notr="true">font-size: 14px;
                                    color: rgba(255, 255, 255, 0.9);
                                    QCheckBox::indicator {
                                    width: 16px;
                                    height: 16px;
                                    border: 1px solid #2ecc71;
                                    border-radius: 3px;
                                    background: rgba(255, 255, 255, 0.1);
                                    }
                                    QCheckBox::indicator:checked {
                                    background: #2ecc71;
                                    border: 1px solid #27ae60;
                                    }</string>
       </property>
       <property name="text">
        <string>自动登录</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="pwdLabel">
     <property name="styleSheet">
      <string notr="true">font-size: 15px;
                            font-weight: bold;
                            color: rgba(255, 255, 255, 0.9);</string>
     </property>
     <property name="text">
      <string>密 码：</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1" colspan="2">
    <widget class="QLineEdit" name="passwordEdit">
     <property name="styleSheet">
      <string notr="true">QLineEdit {
                            height: 36px;
                            border: 1px solid rgba(255,255,255,0.3);
                            border-radius: 6px;
                            padding: 0 12px;
                            background: rgba(255, 255, 255, 0.15);
                            color: white;
                            font-size: 14px;
                            }
                            QLineEdit:focus {
                            border: 1px solid #3498db;
                            }</string>
     </property>
     <property name="echoMode">
      <enum>QLineEdit::EchoMode::Password</enum>
     </property>
     <property name="placeholderText">
      <string>请输入密码</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="3">
    <widget class="QLabel" name="titleLabel">
     <property name="styleSheet">
      <string notr="true">/* 艺术字效果 - 使用Qt支持的属性 */
                            font-size: 38px;
                            font-weight: bold;
                            font-family: 'Microsoft YaHei', 'Segoe UI', sans-serif;
                            color: white;
                            background-color: rgba(0, 0, 0, 0.3);
                            border: 2px solid rgba(255, 255, 255, 0.5);
                            border-radius: 8px;
                            padding: 10px 20px;</string>
     </property>
     <property name="text">
      <string>佛爷个聊天室</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
     <property name="margin">
      <number>10</number>
     </property>
    </widget>
   </item>
   <item row="7" column="2" rowspan="2">
    <widget class="QPushButton" name="closeButton">
     <property name="maximumSize">
      <size>
       <width>48</width>
       <height>48</height>
      </size>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton {
                            background: transparent;
                            border-radius: 24px;
                            }
                            QPushButton:hover {
                            background: rgba(255,0,0,0.2);
                            }</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="icon/icon.qrc">
       <normaloff>:/icon/close.png</normaloff>:/icon/close.png</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>32</width>
       <height>32</height>
      </size>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="3">
    <widget class="QLabel" name="artIcon">
     <property name="minimumSize">
      <size>
       <width>80</width>
       <height>80</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>80</width>
       <height>80</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true">background: transparent;</string>
     </property>
     <property name="pixmap">
      <pixmap>:/icon/chat.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="6" column="2">
    <widget class="QPushButton" name="forgotButton">
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="styleSheet">
      <string notr="true">QPushButton {
                            color: #3498db;
                            text-decoration: underline;
                            border: none;
                            font-size: 13px;
                            font-weight: bold;
                            background: transparent;
                            }
                            QPushButton:hover {
                            color: #e74c3c;
                            }</string>
     </property>
     <property name="text">
      <string>忘记密码?</string>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>30</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="5" column="0">
    <widget class="QLineEdit" name="serverIP">
     <property name="styleSheet">
      <string notr="true">QLineEdit {
                            height: 36px;
                            border: 1px solid rgba(255,255,255,0.3);
                            border-radius: 6px;
                            padding: 0 12px;
                            background: rgba(255, 255, 255, 0.15);
                            color: white;
                            font-size: 14px;
                            }
                            QLineEdit:focus {
                            border: 1px solid #3498db;
                            }</string>
     </property>
     <property name="placeholderText">
      <string>请输入服务器IP</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="icon/icon.qrc"/>
 </resources>
 <connections/>
</ui>
